src/gurt/telemetry.c

```diff
@@ -4288,28 +4287,28 @@ validate_shmem_ptr(struct d_tm_shmem_hdr *shmem_root, void *ptr)
 }
 
 /**
- * Convert the virtual address of the pointer in shared memory from a server
- * address to a client side virtual address.
+ * Convert the virtual address of the pointer in shared memory from a producer
+ * address to a consumer side virtual address.
  *
  * \param[in]	ctx	Telemetry context
  * \param[in]	node	Node this pointer belongs to
  * \param[in]	ptr	The pointer to convert
  *
- * \return		A pointer to the item in the client's address
+ * \return		A pointer to the item in the consumer's address
  *			space
  *			NULL if the pointer is invalid
  */
 void *
 d_tm_conv_ptr(struct d_tm_context *ctx, struct d_tm_node_t *node, void *ptr)
```

Comment:
An aside, you could probably just return the pointer directly in the non-shared case.